### ISSUE TITLE
message_test fails on x86 after #12140

### ISSUE THREAD

**Issue by jamessan** [vim/vim#12946]
### Steps to reproduce

Build and test in an x86 32-bit environment.  This appeared in the CI for the Debian packaging.  I haven't had a chance to look at it further, but wanted to post the issue so it was known. I'll look into it over the next few days if @cvwillegen isn't able to.

### Expected behaviour

Tests all pass.

### Version of Vim

9.0.1811

### Environment

Debian sid

### Logs and stack traces

```shell
message_test: message_test.c:446: test_vim_snprintf_positional: Assertion `n == 6' failed.
Aborted (core dumped)
```

**Comment by k-takata** [comment]
https://github.com/vim/vim/blob/00d07a97ad750f2d2849a649f85ab1610a0b0866/src/message_test.c#L445-L446
It looks that passing a 64-bit number causes the issue. (`(uvarnumber_T)12`)
`fmt_06pb` is defined as `"%1$0.*2$b"`. Should it be `"%1$0.*2$Lb"` to specify that the 1st parameter is 64-bit?
Line 455 should be also the same.

**Comment by jamessan** [comment]
> `fmt_06pb` is defined as `"%1$0.*2$b"`. Should it be `"%1$0.*2$Lb"` to specify that the 1st parameter is 64-bit?

That results in

```
E1400: Cannot mix positional and non-positional arguments: %1$0.*2$Lbmessage_test: message_test.c:446: test_vim_snprintf_positional: Assertion `n == 6' failed.
Aborted
```

**Comment by jamessan** [comment]
Also, the `fmt_06b` tests work fine on 32-bit system, but fail the `n == 6` assert when changed to use `%06Lb`.

**Comment by cvwillegen** [comment]
@RestorerZ also reported an error using positional arguments. Luckily, that was only a documentation error :-)

This one looks like a code error, I'm looking into it. Changing to ``Lb`` gives an assertion failure on my system, as well...

**Comment by chrisbra** [comment]
I think this was fixed by #12985
